File: libopenxds_base/source/cplusplus/IndexOutOfBoundsException.cpp
/*
 *  Copyright (C) 1997-2009 Daniel Robert Bradley. All rights reserved.
 *
 *  This software is redistributable under the terms of the GNU LGPL
 *  (Lesser General Public License).
 *
 *  I.S.Labs is a registered trademark of Daniel Robert Bradley
 */

#include <openxds.exceptions/IndexOutOfBoundsException.h>
#include <openxds.core.adt/IString.h>
#include <openxds.core.adt.std/StdString.h>
#include <cstdlib>

using namespace openxds;
using namespace openxds::exceptions;
using namespace openxds::core::adt;
using namespace openxds::core::adt::std;

static const char* EXCEPTION_TYPE = "IndexOutOfBoundsException";

//-----------------------------------------------------------------------------
// Constructors
//-----------------------------------------------------------------------------

IndexOutOfBoundsException::IndexOutOfBoundsException() : NoSuchElementException()
{}

IndexOutOfBoundsException::IndexOutOfBoundsException( const char* msg ) : NoSuchElementException( msg )
{}

IndexOutOfBoundsException::IndexOutOfBoundsException( IException* anException, const char* msg ) : NoSuchElementException( anException, msg )
{}

IndexOutOfBoundsException::IndexOutOfBoundsException( long index ) : NoSuchElementException()
{
	this->index = index;
}

IndexOutOfBoundsException::~IndexOutOfBoundsException()
{}

//-----------------------------------------------------------------------------
// Public Methods
//-----------------------------------------------------------------------------

const char*
IndexOutOfBoundsException::getType() const
{
	return EXCEPTION_TYPE;
}

//----------------------------------------------------------------------------- 
// Virtual Methods from Object
//----------------------------------------------------------------------------- 

IException* IndexOutOfBoundsException::copy() const
{
	return this->copyIndexOutOfBoundsException();
}

IndexOutOfBoundsException* IndexOutOfBoundsException::copyIndexOutOfBoundsException() const
{
	if ( this->hasChainedException() )
	{
		IException* copy = this->chainedException->copy();
		const char* msg = this->_message->getChars( this->_message );

		return new IndexOutOfBoundsException( copy, msg );
	} else {
		return new IndexOutOfBoundsException( this->_message->getChars( this->_message ) );
	}
}

long
IndexOutOfBoundsException::getIndex() const
{
	return this->index;
}

File: libopenxds_base/source/java/org/openxds/exceptions/IndexOutOfBoundsException.java
package org.openxds.exceptions;

import org.openxds.IException;

@SuppressWarnings("serial")
public class IndexOutOfBoundsException extends NoSuchElementException
implements org.openxds.IException
{
	private static final String EXCEPTION_TYPE = "IndexOutOfBoundsException";
	private int index = -1;
	
	public IndexOutOfBoundsException( int index )
	{
		this.index = index;
	}

	public IndexOutOfBoundsException( java.lang.String msg )
	{
		super( msg );
	}

	public IndexOutOfBoundsException( int index, java.lang.String msg )
	{
		super( msg );
		this.index = index;
	}

	public IndexOutOfBoundsException( IException anException, java.lang.String msg )
	{
		super( anException, msg );
	}
	
	public IndexOutOfBoundsException( int index, IException anException, java.lang.String msg )
	{
		super( anException, msg );
		this.index = index;
	}

	public java.lang.String getType()
	{
		return EXCEPTION_TYPE;
	}

	public IException copy()
	{
		return copyIndexOutOfBoundsException();
	}
	
	public IndexOutOfBoundsException copyIndexOutOfBoundsException()
	{
		if ( hasChainedException() )
		{
			return new IndexOutOfBoundsException( -1, this.chainedException.copy(), this.getMessage() );
		}
		else
		{
			return new IndexOutOfBoundsException( -1, this.getMessage() );
		}
	}
	
	public int getIndex()
	{
		return this.index;
	}
}